### PR TITLE
CT social security benefit adjustment

### PR DESCRIPTION
**Pull request by vrathi101**

### CODE REVIEW

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        p = parameters(
+            period
+        ).gov.states.ct.tax.income.subtractions.social_security
+        filing_status = tax_unit("filing_status", period)
+        ss_benefit = add(tax_unit, period, ["social_security"])
+        agi = add(tax_unit, period, ["adjusted_gross_income"])
+        max_amount = p.max_amount[filing_status]
+        base_amount = p.base_amount[filing_status]
+
+        includable_ss = ss_benefit * 0.25
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:22]: we need to make the 0.25 and 0.5 separate variables

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        excess = agi + 0.5 * ss_benefit - base_amount
+        max_inclusion = min(includable_ss, 0.25 * excess)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:24]:
```suggestion
        max_inclusion = min_(includable_ss, 0.25 * excess)
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        adjusted_ss_benefit = abs(max_inclusion - includable_ss)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:25]:
```suggestion
        adjusted_ss_benefit = max_(max_inclusion - includable_ss, 0)
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        final_ss = where(agi < max_amount, 0, adjusted_ss_benefit)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:26]:
```suggestion
        return where(agi < max_amount, 0, adjusted_ss_benefit)
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        return final_ss
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:27]:
```suggestion
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py (hunk 4 above)
Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:26]: make an eligibility file with the check for agi. If that is how the law defines it

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py]: Usually we are trying to make an additional file for conditions which, if not satisfied, completely exclude the filer from the credit

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:1]: could you make the ct_social_security_benefit_adjustment_eligible.py

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        p = parameters(
+            period
+        ).gov.states.ct.tax.income.subtractions.social_security
+        filing_status = tax_unit("filing_status", period)
+        ss_benefit = add(tax_unit, period, ["social_security"])
+        agi = add(tax_unit, period, ["adjusted_gross_income"])
+        max_amount = p.max_amount[filing_status]
+        base_amount = p.base_amount[filing_status]
+        ss_benefit_frac = 0.25
+        ss_agi_frac = 0.5
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:22]: 0.25 and 0.5 should be parameters

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.ct.tax.income.subtractions.social_security
+        filing_status = tax_unit("filing_status", period)
+        is_eligible = tax_unit(
+            "ct_social_security_benefit_adjustment_eligible", period
+        )
+        ss_benefit = add(tax_unit, period, ["social_security"])
+        agi = add(tax_unit, period, ["adjusted_gross_income"])
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:21]: remove the eligibility code from this file

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment_eligible.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
+from policyengine_us.model_api import *
+
+
+class ct_social_security_benefit_adjustment_eligible(Variable):
+    value_type = float
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment_eligible.py:5]:
```suggestion
    value_type = bool
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment_eligible.py [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    entity = TaxUnit
+    unit = USD
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment_eligible.py:7]:
```suggestion
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment_eligible.py [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    label = "Connecticut social security benefit adjustment eligible"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment_eligible.py:8]:
```suggestion
    label = "Eligible for the Connecticut social security benefit adjustment"
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py [hunk 11]
```diff
@@ -0,0 +1,33 @@
+from policyengine_us.model_api import *
+
+
+class ct_social_security_benefit_adjustment(Variable):
+    value_type = float
+    entity = TaxUnit
+    unit = USD
+    label = "Connecticut social security benefit adjustment"
+    definition_period = YEAR
+    defined_for = StateCode.CT
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:10]:
```suggestion
    defined_for = "ct_social_security_benefit_adjustment_eligible"
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py (hunk 7 above)
Comment (vrathi101) [reply policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:21]: I did the AGI comparison in the eligibility file, I thinks that all we need (?)

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: need git pull upstream master

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/excess_fraction.yaml [hunk 12]
```diff
@@ -0,0 +1,11 @@
+description: Connecticut calculates this percentage of the excess described in Section 86(b)(1) of the Internal Revenue Code.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/excess_fraction.yaml:1]: rephrase to name the variable that is described:
described in Section 86(b)(1) of the Internal Revenue Code.

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/max_amount.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/max_amount.yaml:1]: rename to cap

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/ss_fraction.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/ss_fraction.yaml:1]: rename to rate

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/ss_fraction.yaml [hunk 13]
```diff
@@ -0,0 +1,11 @@
+description: Connecticut calculates social security benefit adjustment with this percentage of benefits and excess.
+values:
+   2019-01-01: 0.25
+metadata:
+  unit: /1
+  label: Connecticut Social Security benefit fraction
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/ss_fraction.yaml:6]:
```suggestion
  label: Connecticut social security benefit subtraction rate
```

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/ss_fraction.yaml [hunk 14]
```diff
@@ -0,0 +1,11 @@
+description: Connecticut calculates social security benefit adjustment with this percentage of benefits and excess.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/ss_fraction.yaml:1]:
```suggestion
description: Connecticut calculates this fraction of social security income under the social security benefit subtarction.
```

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/base_amount.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/base_amount.yaml:1]: is this the max? then rename to cap -

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/max_amount.yaml
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/max_amount.yaml:1]: if this is the maximum income that filers can have to receive the credit

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/base_amount.yaml
Comment (vrathi101) [reply policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/base_amount.yaml:1]: It's actually called "base amount" according to the IRC code (https://www.law.cornell.edu/uscode/text/26/86)

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/excess_fraction.yaml (hunk 12 above)
Comment (vrathi101) [reply policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/excess_fraction.yaml]: In the legal code, its referred to as "percentage of the social security benefits" so I added this

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: need a git pull upstream master

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/base_amount.yaml [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+  period: year
+  unit: currency-USD
+  label: Connecticut social security benefit maximum amount
+  breakdown:
+    - filing_status
+  reference:
+    - title: 2022 Form CT-1040 Connecticut Resident Income Tax Return Instructions
+      href: https://portal.ct.gov/-/media/DRS/Forms/2022/Income/2022-CT-1040-Instructions_1222.pdf#page=8
+    - title: Connecticut General Statutes, Chapter 229, Sec. 12-701, a
+      href: https://www.cga.ct.gov/current/pub/chap_229.htm#sec_12-701a
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/base_amount.yaml:12]: wrong link?

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/base_amount.yaml:12]: all legal references

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/base_amount.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/base_amount.yaml:1]: not sure if I see this parameter in the documents

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/excess_fraction.yaml [hunk 16]
```diff
@@ -0,0 +1,11 @@
+description: Connecticut calculates this percentage of the social security benefits calculated as part of the excess described in Section 86(b)(1) of the Internal Revenue Code.
+values:
+   2019-01-01: 0.50
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/excess_fraction.yaml:3]:
```suggestion
   2019-01-01: 0.5
```

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/excess_fraction.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/excess_fraction.yaml:1]: not sure if I see this parameter either

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+    label = "Connecticut social security benefit adjustment"
+    definition_period = YEAR
+    defined_for = "ct_social_security_benefit_adjustment_eligible"
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.ct.tax.income.subtractions.social_security
+        filing_status = tax_unit("filing_status", period)
+        is_eligible = tax_unit(
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:17]: dont need this as we use in the defined for

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/rate.yaml [hunk 18]
```diff
@@ -0,0 +1,11 @@
+description: Connecticut calculates this fraction of social security income under the social security benefit subtraction.
+values:
+   2019-01-01: 0.25
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/rate.yaml:3]: are there historic values?

Comment (vrathi101) [reply policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/rate.yaml:3]: I don't think so

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/income_threshold.yaml [hunk 19]
```diff
@@ -0,0 +1,23 @@
+description: Connecticut cap the social security benefit subtractions to this amount, based on filing status.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/income_threshold.yaml:1]:
```suggestion
description: Connecticut caps the social security benefit subtractions at this amount, based on filing status.
```

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/income_threshold.yaml [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+metadata:
+  period: year
+  unit: currency-USD
+  label: Connecticut social security benefit maximum amount
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/income_threshold.yaml:5]:
```suggestion
  label: Connecticut social security benefit income threshold
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py [hunk 21]
```diff
@@ -0,0 +1,32 @@
+from policyengine_us.model_api import *
+
+
+class ct_social_security_benefit_adjustment(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:4]:
```suggestion
class ct_social_security_benefit_subtraction(Variable):
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py [hunk 22]
```diff
@@ -0,0 +1,32 @@
...
+    value_type = float
+    entity = TaxUnit
+    unit = USD
+    label = "Connecticut social security benefit adjustment"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:8]:
```suggestion
    label = "Connecticut social security benefit subtraction"
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment_eligible.py [hunk 23]
```diff
@@ -0,0 +1,18 @@
+from policyengine_us.model_api import *
+
+
+class ct_social_security_benefit_adjustment_eligible(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment_eligible.py:4]:
```suggestion
class ct_social_security_benefit_subtarction_eligible(Variable):
```

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/rate.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/rate.yaml:1]: make a subtractions.yaml - similar to the one in the military PR

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py [hunk 24]
```diff
@@ -0,0 +1,32 @@
+from policyengine_us.model_api import *
+
+
+class ct_social_security_benefit_subtraction(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:4]:
```suggestion
class ct_social_security_benefit_adjustment(Variable):
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py [hunk 25]
```diff
@@ -0,0 +1,32 @@
...
+    value_type = float
+    entity = TaxUnit
+    unit = USD
+    label = "Connecticut social security benefit subtraction"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:8]:
```suggestion
    label = "Connecticut social security benefit adjustment"
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_subtraction_eligible.py [hunk 26]
```diff
@@ -0,0 +1,18 @@
+from policyengine_us.model_api import *
+
+
+class ct_social_security_benefit_subtraction_eligible(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_subtraction_eligible.py:4]:
```suggestion
class ct_social_security_benefit_adjustement_eligible(Variable):
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_subtraction_eligible.py [hunk 27]
```diff
@@ -0,0 +1,18 @@
...
+    value_type = bool
+    entity = TaxUnit
+    label = "Eligible for the Connecticut social security benefit subtraction"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_subtraction_eligible.py:7]:
```suggestion
    label = "Eligible for the Connecticut social security benefit adjustment"
```

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/income_threshold.yaml [hunk 28]
```diff
@@ -0,0 +1,23 @@
...
+  period: year
+  unit: currency-USD
+  label: Connecticut social security benefit income threshold
+  breakdown:
+    - filing_status
+  reference:
+    - title: 2022 Form CT-1040 Connecticut Resident Income Tax Return Instructions
+      href: https://portal.ct.gov/-/media/DRS/Forms/2022/Income/2022-CT-1040-Instructions_1222.pdf#page=8
+    - title: Connecticut General Statutes, Chapter 229, Sec. 12-701, a
+      href: https://www.cga.ct.gov/current/pub/chap_229.htm#sec_12-701a
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/income_threshold.yaml:12]: this is the link to tuition subtraction?

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py [hunk 29]
```diff
@@ -0,0 +1,32 @@
...
+        ss_agi_frac = p.rate.lower
+        filing_status = tax_unit("filing_status", period)
+        base_amount = income_base_amount[filing_status]
+        ss_rate = parameters(
+            period
+        ).gov.states.ct.tax.income.subtractions.social_security.rate
+        # is_eligible = tax_unit(
+        #     "ct_social_security_benefit_adjustment_eligible", period
+        # )
+        ss_benefit = add(tax_unit, period, ["social_security"])
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:24]: if this following passage is the right one then it should be `taxable_social_security` 

image

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:1]: Please review the law, I think there are some pieces missing here

image
image

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py [hunk 30]
```diff
@@ -0,0 +1,34 @@
...
+        ss_agi_frac = p.rate.lower
+        filing_status = tax_unit("filing_status", period)
+        base_amount = income_base_amount[filing_status]
+
+        c = parameters(
+            period
+        ).gov.states.ct.tax.income.subtractions.social_security
+        ss_rate = c.rate
+        filing_status = tax_unit("filing_status", period)
+        agi = add(tax_unit, period, ["adjusted_gross_income"])
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:24]:
```suggestion
```

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/rate.yaml [hunk 31]
```diff
@@ -0,0 +1,11 @@
...
+metadata:
+  unit: /1
+  label: Connecticut social security benefit subtraction rate
+  reference:
+    - title: 2022 Form CT-1040 Connecticut Resident Income Tax Return Instructions
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/rate.yaml:8]: add comment stating that tax form does not mention the rate

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/rate.yaml [hunk 32]
```diff
@@ -0,0 +1,11 @@
...
+      href: https://portal.ct.gov/-/media/DRS/Forms/2022/Income/2022-CT-1040-Instructions_1222.pdf#page=8
+    - title: Connecticut General Statutes, Chapter 229, Sec. 12-701, a
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/social_security/rate.yaml:10]: I dont think this is the right section of the law - same for the link

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py [hunk 33]
```diff
@@ -0,0 +1,34 @@
...
+        income_threshold = c.income_threshold[filing_status]
+
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:26]: It seems like the comparison is happening between the `taxable_social_security` and (25% of `social_security` or 25% of a IRS 86 (b) (1) - this is described in tax_unit_taxable_social_security and we need to replicate it (in the irs folder)```

    def formula(tax_unit, period, parameters):
        ss = parameters(period).gov.irs.social_security.taxability
        gross_ss = tax_unit("tax_unit_social_security", period)

        ss_fraction = ss.rate.lower * gross_ss
        modified_agi_plus_half_ss = (
            tax_unit("taxable_ss_magi", period) + ss_fraction
        )
        filing_status = tax_unit("filing_status", period)

        base_amount = ss.threshold.lower[filing_status]

        excess_over_base = max_(0, modified_agi_plus_half_ss - base_amount)

```
![image](https://github.com/PolicyEngine/policyengine-us/assets/110687043/11012d88-0855-4e2a-a354-e6ce003f9ff1)

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: need a git pull upstream master

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_magi_excess_over_base.py [hunk 34]
```diff
@@ -0,0 +1,24 @@
+from policyengine_us.model_api import *
+
+
+class ct_magi_excess_over_base(Variable):
+    value_type = float
+    entity = TaxUnit
+    unit = USD
+    label = "Connecticut excess of modified adjusted gross income and social security over base amount"
+    definition_period = YEAR
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_magi_excess_over_base.py:9]: add a reference

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_magi_excess_over_base.py:9]: from IRS and CT

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_magi_excess_over_base.py [hunk 35]
```diff
@@ -0,0 +1,24 @@
+from policyengine_us.model_api import *
+
+
+class ct_magi_excess_over_base(Variable):
+    value_type = float
+    entity = TaxUnit
+    unit = USD
+    label = "Connecticut excess of modified adjusted gross income and social security over base amount"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_magi_excess_over_base.py:8]:
```suggestion
    label = "Excess of modified adjusted gross income and social security over base amount calculation for the Connecticut social security subtraction"
```

File: changelog_entry.yaml
Comment (vrathi101) [reply changelog_entry.yaml:1]: Am not able to (forgot to ask you about this during meeting), getting:
`fatal: 'upstream' does not appear to be a git repository
fatal: Could not read from remote repository.`

File: policyengine_us/parameters/gov/states/ct/tax/income/subtractions/subtractions.yaml [hunk 36]
```diff
@@ -1,12 +1,11 @@
 description: Connecticut counts these sources as subtractions.
 values:
   2021-01-01:
-    - military_retirement_pay
+    - ct_social_security_benefit_adjustment
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ct/tax/income/subtractions/subtractions.yaml:4]:
```suggestion
     - military_retirement_pay
     - ct_social_security_benefit_adjustment
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_magi_excess_over_base.py [hunk 37]
```diff
@@ -0,0 +1,30 @@
...
+    reference = (
+        # Connecticut General Statutes, Chapter 229, Sec. 12-701, (20), (b), (x), (iii) and (iv)
+        "https://www.cga.ct.gov/current/pub/chap_229.htm#sec_12-701"
+        # Section 86(b)(1) of the Internal Revenue Code
+        "https://www.law.cornell.edu/uscode/text/26/86"
+    )
+    definition_period = YEAR
+    defined_for = StateCode.CT
+
+    def formula(tax_unit, period, parameters):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_magi_excess_over_base.py:18]:
```suggestion
    # Code extracted from tax_unit_taxable_social_security
    def formula(tax_unit, period, parameters):
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py [hunk 38]
```diff
@@ -0,0 +1,26 @@
...
+    definition_period = YEAR
+    defined_for = StateCode.CT
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.ct.tax.income.subtractions.social_security
+        filing_status = tax_unit("filing_status", period)
+        ss_rate = p.rate
+        agi = add(tax_unit, period, ["adjusted_gross_income"])
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:18]:
```suggestion
        agi = tax_unit ("adjusted_gross_income", period)
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py [hunk 39]
```diff
@@ -0,0 +1,26 @@
...
+        income_threshold = p.income_threshold[filing_status]
+        ss_benefit = add(tax_unit, period, ["taxable_social_security"])
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:20]:
```suggestion
        taxable_ss = add(tax_unit, period, ["taxable_social_security"])
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py [hunk 40]
```diff
@@ -0,0 +1,26 @@
...
+        excess = tax_unit("ct_magi_excess_over_base", period)
+
+        includable_ss = ss_benefit * ss_rate
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:23]:
```suggestion
        total_ss = add(tax_unit, period, ["social_security")
        ss_fraction = total_ss * ss_rate
```

File: policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py [hunk 41]
```diff
@@ -0,0 +1,26 @@
...
+        max_inclusion = min_(includable_ss, ss_rate * excess)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ct/tax/income/subtractions/ct_social_security_benefit_adjustment.py:24]:
```suggestion
        max_inclusion = min_(ss_fraction, ss_rate * excess)
```

**Review by PavelMakarchuk — APPROVED** [review]
Approved after the changes

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**